omod-2.4/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/MainResourceControllerTest.java

```diff
@@ -9,9 +9,11 @@
  */
 package org.openmrs.module.webservices.rest.web.v1_0.controller;
```

Comment:
I went directly,  I don't know if this class is use out of openmrs

Reply:
what do you mean by you went directly?

Reply:
> what do you mean by you went directly?

I did not duplicate the class . It an abstract class .

Reply:
did you do a search within the codebase and one on github for example within the org:openmrs ?

Reply:
and why did you choose it then? is there no unit test file the does not extend such an abstract test class in this module?

Reply:
Yes @teleivo  there is only one unit test that extend this class. can i duplicate this class  since that is only one unit test that extend it  ?